modules/spacelift-automation/variables.tf

```diff
@@ -0,0 +1,242 @@
+# GitHub
+variable "github_enterprise" {
+  type = object({
+    namespace = string
+    id        = optional(string)
+  })
+  description = "The GitHub VCS settings"
+}
+
+variable "repository" {
+  type        = string
+  description = "The name of your infrastructure repo"
+}
+
+variable "branch" {
+  type        = string
+  description = "Specify which branch to use within the infrastructure repository."
+  default     = "main"
+}
+
+variable "root_modules_path" {
+  type        = string
+  description = "The path, relative to the root of the repository, where the root module can be found."
+  default     = "root-modules"
+}
+
+variable "enabled_root_modules" {
```

Comment:
Is there a way to enable ALL root modules?

Reply:
No, will add.

Reply:
@Gowiem I added variable `enable_all_root_modules` to support this. 
TF has no built-in function to list directories directly, so I also had to add an external data source to list the root modules without ugly hacks. If you know an elegant way to do this - I'd love to learn it.

Reply:
@gberenice Check out the untested pseudo code below, I think we can do it this way. Let me know if this feels too hacky: 

```
_all_stack_files = fileset(var.root_modules_path, "**/stacks/*.yaml")

root_module_names = distinct([ for file in local._all_stack_files: split(file, "/stacks/")[0] ])
```

Wanna take my shitty code and see if that'll work?

Reply:
I was thinking about that. What I didn't like was doing tons of replace/split/distinct, but eventually, this probably makes more sense as we iterate over directories that 100% contain stack yaml files. 
Updated with similar code to yours above + removed external data source usage.